### DISCUSSION TITLE
use chakra factory to pass button props

### DISCUSSION THREAD

**Discussion by peersky** [chakra-ui/chakra-ui#4586] [Q&A]
I have this code that wraps chakra button: 

```js
const _RouteButton = (props) => {
  return (
    <NextLink href={props.href} passHref>
      <Button {...props}>{props.children}</Button>
    </NextLink>
  );
};

const RouteButton = chakra(_RouteButton, "button");

export default RouteButton;

```

Is it possible to specify for chakra ui that only butten properties are valid, and also that button properties appear in autosuggest of IDE ? 

Thanx
Peersky

**Comment by payapula** [comment]
Hi @peersky 

### If this is typescript, I would suggest this code,

```jsx
import { Button, ComponentWithAs } from '@chakra-ui/react';
import Link from 'next/link';
import React from 'react';

interface LinkbuttonProps {
    href: string;
    children: React.ReactNode;
}

export const Linkbutton: ComponentWithAs<'button', LinkbuttonProps> = ({
    children,
    href,
    ...props
}) => {
    return (
        <Link href={href} passHref>
            <Button as="a" {...props}>  // Notice as="a" here which renders <a> instead of <button>
                {children}
            </Button>
        </Link>
    );
};

```

`ComponentWithAs` -> Defines the props for this component, 'button' means default button props, along with that we need `LinkButtonProps`

### If you are asking for Plain Javascript,

I would recommend this article and JSDoc solution - https://dev.to/maxbvrn/react-props-auto-complete-in-vs-code-2ana
